components/benefit_sponsors/app/models/benefit_sponsors/concerns/employer_datatable_concern.rb

```diff
@@ -26,7 +26,9 @@ module EmployerDatatableConcern
           orgs =  BenefitSponsors::Organizations::Organization.employer_profiles.where(:"profiles.employer_attestation.aasm_state".in => EmployerAttestation::ATTESTATION_KINDS)
           self.where(:"organization".in => orgs.collect{|org| org.id.to_s}) }
 
-        scope :benefit_sponsorship_applicant, -> () { where(:"aasm_state" => :applicant) }
+        scope :benefit_sponsorship_applicant, -> () { 
```

Comment:
This is the logic that needed to be fixed to only display draft/applicant employers after clicking the applicants filter.